src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Interactivity API: WP_Interactivity_API_Directives_Processor class.
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ * @since 6.5.0
+ */
+
+/**
+ * Class used to iterate over the tags of an HTML string and help process the
+ * directive attributes.
+ *
+ * @since 6.5.0
+ *
+ * @access private
+ */
+class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Processor {
+	/**
+	 * List of thags whose closer tag is not visited by the WP_HTML_Tag_Processor.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string[]
+	 */
+	const TAGS_THAT_DONT_VISIT_CLOSER_TAG = array(
+		'SCRIPT',
+		'IFRAME',
+		'NOEMBED',
+		'NOFRAMES',
+		'STYLE',
+		'TEXTAREA',
+		'TITLE',
+		'XMP',
+	);
+
+	/**
+	 * Returns the content between two balanced tags.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @access private
+	 *
+	 * @return string|null The content between the current opening and its matching closing tag or null if it doesn't
+	 *                     find the matching closing tag.
+	 */
+	public function get_content_between_balanced_tags() {
+		$bookmarks = $this->get_balanced_tag_bookmarks();
+		if ( ! $bookmarks ) {
+			return null;
+		}
+		list( $start_name, $end_name ) = $bookmarks;
+
+		$start = $this->bookmarks[ $start_name ]->start + $this->bookmarks[ $start_name ]->length + 1;
+		$end   = $this->bookmarks[ $end_name ]->start;
+
+		$this->seek( $start_name );
+		$this->release_bookmark( $start_name );
+		$this->release_bookmark( $end_name );
+
+		return substr( $this->html, $start, $end - $start );
+	}
+
+	/**
+	 * Sets the content between two balanced tags.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @access private
+	 *
+	 * @param string $new_content The string to replace the content between the matching tags.
+	 * @return bool Whether the content was successfully replaced.
+	 */
+	public function set_content_between_balanced_tags( string $new_content ): bool {
+		$this->get_updated_html();
+
+		$bookmarks = $this->get_balanced_tag_bookmarks();
+		if ( ! $bookmarks ) {
+			return false;
+		}
+		list( $start_name, $end_name ) = $bookmarks;
+
+		$start = $this->bookmarks[ $start_name ]->start + $this->bookmarks[ $start_name ]->length + 1;
+		$end   = $this->bookmarks[ $end_name ]->start;
+
+		$this->seek( $start_name );
+		$this->release_bookmark( $start_name );
+		$this->release_bookmark( $end_name );
+
+		$this->lexical_updates[] = new WP_HTML_Text_Replacement( $start, $end - $start, esc_html( $new_content ) );
+		return true;
+	}
```

Comment:
Let me merge the `data-wp-each` processor first, because it includes a couple of changes to those functions. I'll keep this unresolved until then to remind me to try it.

Reply:
I've added this function in https://github.com/WordPress/wordpress-develop/pull/5953/commits/4618663b3575d0d074fdbaf512c29c05ed600e6e.